src/lib/writer-json-sarif.cc

```diff
@@ -258,6 +258,26 @@ static void sarifEncodeEvt(array *pDst, const Defect &def, unsigned idx)
     pDst->push_back(std::move(tfLoc));
 }
 
+void sarifEncodeSnippet(object *reg, const std::string &msg)
+{
+    // check whether the "snippet" node already exists
+    value &valSnip = (*reg)["snippet"];
+    if (!valSnip.is_object()) {
+        // create the "text" node containing the header line
+        object objText = {{ "text", "Problem detected in this context:" }};
+        valSnip = objText;
```

Comment:
You can use `emplaceObject` to create and access a object in place: https://www.boost.org/doc/libs/1_79_0/libs/json/doc/html/json/ref/boost__json__value/emplace_object.html

```suggestion
        valSnip.emplaceObject() = {{ "text", "Problem detected in this context:" }};
```

Reply:
Addressed in commit 3adeff9a317a808cb932c51bbc809676db06b62e.